### PR TITLE
autoupdate: adding auto-update test back

### PR DESCRIPTION
**Pull request by karthikprabhuvinod**
Signed-off-by: Karthik Prabhu Vinod karthik.prabhu.vinod@intel.com

### CODE REVIEW

**Comment by karthikprabhuvinod** [comment]
https://github.com/clearlinux/swupd-client/issues/1120

File: test/functional/autoupdate/aup-basic.bats [hunk 1]
```diff
@@ -59,3 +52,45 @@ test_teardown() {
 	assert_status_is_not "$SWUPD_OK"
 
 }
+
+@test "AUT002: Check auto-update status in different root" {
```
Comment (castulo) [inline test/functional/autoupdate/aup-basic.bats]: aren't you doing basically the same thing as the previous test in this one?

File: test/functional/autoupdate/aup-basic.bats [hunk 2]
```diff
@@ -7,26 +7,19 @@ load "../testlib"
...
-	if [ -z "$TRAVIS" ]; then
-		return
-	fi
-
 	destroy_test_environment "$TEST_NAME"
 
 }
 
-@test "AUT001: Basic test, Check auto-update enable" {
+@test "AUT001: auto-update disable-enable in different root" {
```
Comment (castulo) [inline test/functional/autoupdate/aup-basic.bats]: why did you add "in different root"? because we are using a `--path`?

File: test/functional/autoupdate/aup-basic.bats (hunk 1 above)
Comment (karthikprabhuvinod) [reply test/functional/autoupdate/aup-basic.bats]: So the second one is exclusively for checking the status of auto-update. I created a different test because the previous test was getting too big

File: test/functional/autoupdate/aup-basic.bats (hunk 2 above)
Comment (karthikprabhuvinod) [reply test/functional/autoupdate/aup-basic.bats]: Yes...should I use a different name?

Comment (castulo) [reply test/functional/autoupdate/aup-basic.bats]: all our tests (or almost all) use --path, so I would just not specify that at all.

File: test/functional/autoupdate/aup-basic.bats (hunk 1 above)
Comment (castulo) [reply test/functional/autoupdate/aup-basic.bats]: but you are still doing the same stuff from the first test plus some more stuff, so I don't see how that is going to help with that....

Ideally every test should be atomic, so AUT001 should ideally be split into two tests, one which enables autoupdate (from an initially disabled system) and another one which disables it (from an initially enabled system). The same thing for the autoupdate tests you are writing (for checking the status), there should be two tests, one for checking the status when the system has autoupdate disabled and one when is enabled. Those conditions, the ones I'm referring to when saying "initially" are supposed to be set up in the test_setup() function, so the tests would be split into two or more test files.

Having said that, if you still just want to add that other verification I would just do it in the same test (AUT001), the only thing you are not doing from AUT001 in AUT002 is to check the existence of files:
```
# check for files(should exist and point to /dev/null)
run sudo sh -c "test -c $PATH_PREFIX/etc/systemd/system/swupd-update.service"
assert_status_is "$SWUPD_OK"
run sudo sh -c "test -c $PATH_PREFIX/etc/systemd/system/swupd-update.timer"
assert_status_is "$SWUPD_OK"
```
So you could just add your verifications to this test and maybe change the code above with something like this"
```
assert_file_exists $PATH_PREFIX/etc/systemd/system/swupd-update.service
assert_file_exists $PATH_PREFIX/etc/systemd/system/swupd-update.timer
```
At least that way you are saving a couple of lines of code, plus you would be using the assertions from the testlib which is supposed to be the correct way of verifying results.

File: test/functional/autoupdate/aup-basic.bats (hunk 2 above)
Comment (karthikprabhuvinod) [reply test/functional/autoupdate/aup-basic.bats]: Okay will change that

File: test/functional/autoupdate/aup-basic.bats (hunk 1 above)
Comment (karthikprabhuvinod) [reply test/functional/autoupdate/aup-basic.bats]: Makes sense. Okay I will add the verifications to the same test.

for the assert_file_exists i remember you asking of this change before..The problem these files are charcters files as they point to /dev/null and assert_file_exists. I will change the assert_file_exist in that case.

Comment (castulo) [reply test/functional/autoupdate/aup-basic.bats]: > for the assert_file_exists i remember you asking of this change before..The problem these files are charcters files as they point to /dev/null and assert_file_exists. I will change the assert_file_exist in that case.

Ahh I didn't remember that, but it makes sense that I complaint about that before haha 😝

File: test/functional/autoupdate/aup-basic.bats [hunk 3]
```diff
@@ -53,9 +53,7 @@ test_teardown() {
 	assert_is_output "$expected_output"
 
 	# check for files(should not exist)
-	run sudo sh -c "test -c $PATH_PREFIX/etc/systemd/system/swupd-update.service"
-	assert_status_is_not "$SWUPD_OK"
-	run sudo sh -c "test -c $PATH_PREFIX/etc/systemd/system/swupd-update.timer"
-	assert_status_is_not "$SWUPD_OK"
+	assert_file_not_exists "$PATH_PREFIX/etc/systemd/system/swupd-update.service"
+	assert_file_not_exists "$PATH_PREFIX/etc/systemd/system/swupd-update.timer"
```
Comment (castulo) [inline test/functional/autoupdate/aup-basic.bats]: don't you also need to run "autoupdate" again to check for the correct status after enabling it?

Comment (karthikprabhuvinod) [reply test/functional/autoupdate/aup-basic.bats]: I didn't add it initially as it would have created confusion. But I see why not. Please let me know if that explanation looks good for it

Comment (castulo) [reply test/functional/autoupdate/aup-basic.bats]: oh so there is not way to use autoupdate to get the status when using a --path? I'm confused now 😛

File: scripts/build_ci.bash [hunk 4]
```diff
@@ -10,6 +10,8 @@ sudo apt-get install check
 sudo apt-get install bats
 sudo pip install coverxygen
 sudo ln -s /bin/systemctl /usr/bin/systemctl
+sudo systemctl mask swupd-update.service
+sudo systemctl mask swupd-update.timer
```
Comment (otaviobp) [inline scripts/build_ci.bash]: If you mask the timer and service you aren't testing if the disabled is really working.
I checked the current error and systemctl is falling with message:

```
External command: Created symlink /home/runner/work/swupd-client/swupd-client/aup-basic_AUT001/testfs/target-dir/etc/systemd/system/swupd-update.service → /dev/null.
External command: Created symlink /home/runner/work/swupd-client/swupd-client/aup-basic_AUT001/testfs/target-dir/etc/systemd/system/swupd-update.timer → /dev/null.
External command: Failed to stop swupd-update.service: Unit swupd-update.service not loaded.
External command: Failed to stop swupd-update.timer: Unit swupd-update.timer not loaded.
```

Did you try to initialize the service instead of masking it?

Comment (karthikprabhuvinod) [reply scripts/build_ci.bash]: @otaviobp : You were right...placing starting the service unit files inside and starting the swupd-update.timer helped it. I did need the mask.
